models/maat/R/model2netcdf.MAAT.R

```diff
@@ -50,6 +31,37 @@ model2netcdf.MAAT <- function(rundir, outdir, sitelat = -999, sitelon = -999, st
   # setup constants
   day_secs <- udunits2::ud.convert(1, "day", "seconds")
   
+  # setup helper function
+  var_update <- function(data, out, oldname, newname, oldunits, newunits=NULL, missval=-999, longname, ncdims) {
+    # ifelse is no longer working as expected, so now we have this function to deal with any Inf values
+    f_sort <- function(s) {
+      if (is.infinite(s)) -999
+      else try(PEcAn.utils::misc.convert(s, oldunits, newunits), silent = TRUE)
```

Comment:
What is the expected behavior here if `s` is a number but the `misc.convert` fails? Right now, this will return a `try-error` object, which I suspect downstream code will not know how to handle. (If it can, feel free to silently ignore this comment).

I would suggest the following more explicit form that returns the object in its original form, perhaps with an informative `logger.warn` call thrown in.
```suggestion
      else tryCatch(
        PEcAn.utils::misc.convert(s, oldunits, newunits),
        error = function(e) s
      )
```

Reply:
I will need to test this first